src/formatDate.js

```diff
@@ -50,7 +50,43 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitDate = date.split(fromFormat[3]);
+  const fromMp = splitDate.reduce(
+    (acc, nextVal, index) => ({ ...acc,
+      [fromFormat[index].charAt(0)]: nextVal }),
+    {}
+  );
+
+  const resultArray = [];
+
+  for (const key of toFormat) {
+    const keyChar = key.charAt(0);
+
+    switch (keyChar) {
+      case 'D':
+        resultArray.push(fromMp[keyChar]);
```

Comment:
seems case 'D' and 'M' doing the same operation, please join they into one to not duplicate code